packages/react-router/modules/matchPath.js

```diff
@@ -26,15 +26,19 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, options = {}) => {
+const matchPath = (pathname, options = {}, parentMatch = null) => {
   if (typeof options === 'string')
     options = { path: options }
 
   const { exact = false, strict = false } = options
-  const path = options.path || options.from
+  let path = options.path != null ? options.path : options.from
+  const absolute = isAbsolute(path)
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  if (path == null)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
+
+  if (!absolute)
+    path = resolvePath(path, parentMatch && parentMatch.url)
```

Comment:
Hmm, you're right. I'll have to think more about this.

Right now there is no root `match`, but it is possible that if there was (i.e., `{ path: '/', url: '/' }`), then `parentMatch === null` would only happen in this situation and it would be easy to just return `null` when the parent match is also null.

```js
if (parentMatch === null)
  return null
```